specs/emip-004.md

```diff
@@ -0,0 +1,67 @@
+
+# Background
+
+[BIP32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#Specification_Key_derivation) introduced the concept of a hierarchical deterministic (HD) wallet, a logical hierarchy structure built using child key derivation functions (CKD) that allows to generate multiple wallets from a master seed. 
+
+![image](https://raw.githubusercontent.com/bitcoin/bips/master/bip-0032/derivation.png)
+
+[BIP43](https://github.com/bitcoin/bips/blob/master/bip-0043.mediawiki) added a purpose field for derivations (derivation path) and [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) introduced the concept of multi-account hierarchy deterministic wallets, defining 5 leves of derivation.
+
+`m / purpose' / coin_type' / account' / change / address_index`
+
+You can find a list of registered purposes [here](https://wiki.trezor.io/Cryptocurrency_standards). Yoroi is an example of a HD wallet following the BIP44 standard. Yoroi addresses use the following derivation path:
+
+`m / 44' / 1815' / 0' / 0 / 1`
+
+In this example derivation, the purpose is 44 (multi-account HD wallet), the coin type is 1815 (Cardano), the account index is 0 (first account), the change index is also 0, and the address index is 1 (second address in the account).
+
+# Master seed and Private chain
+
+There are three possibilites when considering the availability of the wallet account master key pair (which from now on we call the *account key pair*):
+
+* **Standard Yoroi wallets**: Both the public and the private keys of the account key pair are available.
+* **Hardware wallets**: Only the public key is available.
+* **Balance tracker wallets**: No keys are available.
+
+Based on the above, we identify two scenarios:
+
+* **S1**: account public key is available
+* **S2**: account public key is not available
+
+In both cases, even if the account public key is available, we don't derive the child keys directly from the account as we don't want to conflict with any other derivations. We discuss each scenario in the following sections. 
+
+## Public key is available
+
+We create a new key pair from the account public key in the following way:
+
+* Hash the account public key with `Blake2b` and specify an output length of 32 bytes.
+* Use the above result as a seed to create a mnemonic: `bip39.entropyToMnemonic(blakeHash)`.
+* Create entropy from the english mnemonic obtained in the previous step, using the Cardano WASM bindings: `RustModule.Wallet.Entropy.from_english_mnemonics(mnemonic)`.
+* Create a new private key with the previous result: `RustModule.PrivateKey.new(entropy)`
```

Comment:
@SebastienGllmt, does this new child private key locate on hardware (in case of hardware wallet)?

Reply:
You don't have to interact with the hardware wallet for this specification by design because otherwise the user (for example) would have to use their hardware wallet every time they want add a memo to their transaction (would not be very good UX)
So the hardware wallet is oblivious to this whole utility key derivation scheme by design

Reply:
It's clear. Thank you for the explanation